### PR TITLE
chore: security helper with list of should-be-public urls

### PR DESCRIPTION
**Pull request by joheriks**
Part of vaadin/spring#678

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/internal/SecurityHelper.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    private SecurityHelper() {
+    }
+
+    /**
+     * URLs matching these patterns should be publicly available for application
+     * to work. Used for defining bypass rules in e.g. Spring Security.
+     */
+    public static String[] PUBLIC_VAADIN_URLS = {
+            "/" + Constants.VAADIN_MAPPING + "/**",
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/internal/SecurityHelper.java:41]: Is it a problem that /VAADIN/dynamic is public? What does it contain?

Comment (knoobie) [reply flow-server/src/main/java/com/vaadin/flow/internal/SecurityHelper.java:41]: In the spring pr is 

```java
                .filter(pattern -> !pattern.equals("/VAADIN/**"))
```

to remove the constant again. Not sure if that's the best way.

Comment (joheriks) [reply flow-server/src/main/java/com/vaadin/flow/internal/SecurityHelper.java]: Removed `/VAADIN/**` from `PUBLIC_VAADIN_URLS`.

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]
The code looks good to me, the sonar issues are reasonable, could be fixed.

**Comment by joheriks** [comment]
> The code looks good to me, the sonar issues are reasonable, could be fixed.

Sonar issues fixed and validation build now passed.

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**